### PR TITLE
Edgelet RUSTSEC dep update

### PR DESCRIPTION
**Pull request by nyanzebra**
Updates deps where possible to avoid `cargo audit` errors. Currently, edgelet is using a parse_duration crate that has a security vulnerability but no safe upgrade! So, FYI on that...

### CODE REVIEW

File: edgelet/edgelet-docker/src/settings.rs [hunk 1]
```diff
@@ -308,7 +308,7 @@ impl From<ErrorKind> for LoadSettingsError {
 
 #[cfg(test)]
 mod tests {
-    #[cfg(target_os = "linux")]
+    #[cfg(unix)]
```
Comment (arsing) [inline edgelet/edgelet-docker/src/settings.rs:311]: If you're changing it you can just remove it entirely. All the code in this branch is Linux-only now.

**Review by arsing — CHANGES_REQUESTED** [review]

**Comment by arsing** [comment]
What RUSTSEC is it specifically?

File: edgelet/edgelet-docker/src/settings.rs (hunk 1 above)
Comment (nyanzebra) [reply edgelet/edgelet-docker/src/settings.rs:311]: Sweet, all in favour of removing unnecessary conditional compilation :)

**Comment by nyanzebra** [comment]
> 
> 
> What RUSTSEC is it specifically?

Do you mean the list of of errors from `cargo audit`? If so this is the list (wall of text incoming):

```
C:\Users\Robert\Documents\GitHub\iotedge\edgelet>cargo audit
    Fetching advisory database from `https://github.com/RustSec/advisory-db.git`
      Loaded 302 security advisories (from C:\Users\Robert\.cargo\advisory-db)
    Updating crates.io index
    Scanning Cargo.lock for vulnerabilities (290 crate dependencies)
Crate:         arc-swap
Version:       0.4.7
Title:         Dangling reference in `access::Map` with Constant
Date:          2020-12-10
ID:            RUSTSEC-2020-0091
URL:           https://rustsec.org/advisories/RUSTSEC-2020-0091
Solution:      Upgrade to >=1.1.0 OR >=0.4.8
Dependency tree:
arc-swap 0.4.7
└── signal-hook-registry 1.2.1
    └── tokio-signal 0.2.9
        ├── tokio-process 0.2.5
        │   └── edgelet-docker 0.1.0
        │       ├── iotedge 0.1.0
        │       ├── edgelet-http-mgmt 0.1.0
        │       │   ├── iotedge 0.1.0
        │       │   ├── edgelet-http-workload 0.1.0
        │       │   │   └── aziot-edged 0.1.0
        │       │   └── aziot-edged 0.1.0
        │       └── aziot-edged 0.1.0
        └── aziot-edged 0.1.0

Crate:         futures-task
Version:       0.3.5
Title:         futures_task::waker may cause a use-after-free if used on a type that isn't 'static
Date:          2020-09-04
ID:            RUSTSEC-2020-0060
URL:           https://rustsec.org/advisories/RUSTSEC-2020-0060
Solution:      Upgrade to >=0.3.6
Dependency tree:
futures-task 0.3.5
└── futures-util 0.3.5
    └── http-common 0.1.0
        ├── edgelet-http-workload 0.1.0
        │   └── aziot-edged 0.1.0
        ├── aziotctl-common 0.1.0
        │   └── iotedge 0.1.0
        ├── aziot-tpmd-config 0.1.0
        │   ├── iotedge 0.1.0
        │   └── aziotctl-common 0.1.0
        ├── aziot-keyd-config 0.1.0
        │   ├── iotedge 0.1.0
        │   └── aziotctl-common 0.1.0
        ├── aziot-key-common-http 0.1.0
        │   ├── edgelet-http-workload 0.1.0
        │   ├── aziot-key-client 0.1.0
        │   │   ├── edgelet-http-workload 0.1.0
        │   │   ├── aziot-key-openssl-engine 0.1.0
        │   │   │   ├── edgelet-http-workload 0.1.0
        │   │   │   └── aziot-edged 0.1.0
        │   │   └── aziot-edged 0.1.0
        │   └── aziot-edged 0.1.0
        ├── aziot-key-client 0.1.0
        ├── aziot-identityd-config 0.1.0
        │   ├── iotedge 0.1.0
        │   └── aziotctl-common 0.1.0
        ├── aziot-identity-common-http 0.1.0
        │   ├── iotedge 0.1.0
        │   ├── identity-client 0.1.0
        │   │   ├── iotedge 0.1.0
        │   │   ├── edgelet-http-workload 0.1.0
        │   │   ├── edgelet-http-mgmt 0.1.0
        │   │   │   ├── iotedge 0.1.0
        │   │   │   ├── edgelet-http-workload 0.1.0
        │   │   │   └── aziot-edged 0.1.0
        │   │   └── aziot-edged 0.1.0
        │   └── aziot-edged 0.1.0
        ├── aziot-identity-common 0.1.0
        │   ├── iotedge 0.1.0
        │   ├── identity-client 0.1.0
        │   ├── edgelet-http-workload 0.1.0
        │   ├── edgelet-http-mgmt 0.1.0
        │   ├── aziot-identityd-config 0.1.0
        │   ├── aziot-identity-common-http 0.1.0
        │   └── aziot-edged 0.1.0
        ├── aziot-edged 0.1.0
        └── aziot-certd-config 0.1.0
            ├── iotedge 0.1.0
            └── aziotctl-common 0.1.0

Crate:         futures-util
Version:       0.3.5
Title:         MutexGuard::map can cause a data race in safe code
Date:          2020-10-22
ID:            RUSTSEC-2020-0059
URL:           https://rustsec.org/advisories/RUSTSEC-2020-0059
Solution:      Upgrade to >=0.3.7
Dependency tree:
futures-util 0.3.5
└── http-common 0.1.0
    ├── edgelet-http-workload 0.1.0
    │   └── aziot-edged 0.1.0
    ├── aziotctl-common 0.1.0
    │   └── iotedge 0.1.0
    ├── aziot-tpmd-config 0.1.0
    │   ├── iotedge 0.1.0
    │   └── aziotctl-common 0.1.0
    ├── aziot-keyd-config 0.1.0
    │   ├── iotedge 0.1.0
    │   └── aziotctl-common 0.1.0
    ├── aziot-key-common-http 0.1.0
    │   ├── edgelet-http-workload 0.1.0
    │   ├── aziot-key-client 0.1.0
    │   │   ├── edgelet-http-workload 0.1.0
    │   │   ├── aziot-key-openssl-engine 0.1.0
    │   │   │   ├── edgelet-http-workload 0.1.0
    │   │   │   └── aziot-edged 0.1.0
    │   │   └── aziot-edged 0.1.0
    │   └── aziot-edged 0.1.0
    ├── aziot-key-client 0.1.0
    ├── aziot-identityd-config 0.1.0
    │   ├── iotedge 0.1.0
    │   └── aziotctl-common 0.1.0
    ├── aziot-identity-common-http 0.1.0
    │   ├── iotedge 0.1.0
    │   ├── identity-client 0.1.0
    │   │   ├── iotedge 0.1.0
    │   │   ├── edgelet-http-workload 0.1.0
    │   │   ├── edgelet-http-mgmt 0.1.0
    │   │   │   ├── iotedge 0.1.0
    │   │   │   ├── edgelet-http-workload 0.1.0
    │   │   │   └── aziot-edged 0.1.0
    │   │   └── aziot-edged 0.1.0
    │   └── aziot-edged 0.1.0
    ├── aziot-identity-common 0.1.0
    │   ├── iotedge 0.1.0
    │   ├── identity-client 0.1.0
    │   ├── edgelet-http-workload 0.1.0
    │   ├── edgelet-http-mgmt 0.1.0
    │   ├── aziot-identityd-config 0.1.0
    │   ├── aziot-identity-common-http 0.1.0
    │   └── aziot-edged 0.1.0
    ├── aziot-edged 0.1.0
    └── aziot-certd-config 0.1.0
        ├── iotedge 0.1.0
        └── aziotctl-common 0.1.0

Crate:         generic-array
Version:       0.9.0
Title:         arr! macro erases lifetimes
Date:          2020-04-09
ID:            RUSTSEC-2020-0146
URL:           https://rustsec.org/advisories/RUSTSEC-2020-0146
Solution:      Upgrade to >=0.8.4, <0.9.0 OR >=0.9.1, <0.10.0 OR >=0.10.1, <0.11.0 OR >=0.11.2, <0.12.0 OR >=0.12.4, <0.13.0 OR >=0.13.3
Dependency tree:
generic-array 0.9.0
├── digest 0.7.6
│   ├── sha2 0.7.1
│   │   ├── edgelet-core 0.1.0
│   │   │   ├── support-bundle 0.1.0
│   │   │   │   ├── iotedge 0.1.0
│   │   │   │   └── edgelet-http-mgmt 0.1.0
│   │   │   │       ├── iotedge 0.1.0
│   │   │   │       ├── edgelet-http-workload 0.1.0
│   │   │   │       │   └── aziot-edged 0.1.0
│   │   │   │       └── aziot-edged 0.1.0
│   │   │   ├── iotedge 0.1.0
│   │   │   ├── identity-client 0.1.0
│   │   │   │   ├── iotedge 0.1.0
│   │   │   │   ├── edgelet-http-workload 0.1.0
│   │   │   │   ├── edgelet-http-mgmt 0.1.0
│   │   │   │   └── aziot-edged 0.1.0
│   │   │   ├── edgelet-test-utils 0.1.0
│   │   │   │   ├── support-bundle 0.1.0
│   │   │   │   ├── iotedge 0.1.0
│   │   │   │   ├── edgelet-http-workload 0.1.0
│   │   │   │   ├── edgelet-http-mgmt 0.1.0
│   │   │   │   ├── edgelet-http 0.1.0
│   │   │   │   │   ├── iotedge 0.1.0
│   │   │   │   │   ├── identity-client 0.1.0
│   │   │   │   │   ├── edgelet-http-workload 0.1.0
│   │   │   │   │   ├── edgelet-http-mgmt 0.1.0
│   │   │   │   │   ├── edgelet-docker 0.1.0
│   │   │   │   │   │   ├── iotedge 0.1.0
│   │   │   │   │   │   ├── edgelet-http-mgmt 0.1.0
│   │   │   │   │   │   └── aziot-edged 0.1.0
│   │   │   │   │   ├── cert-client 0.1.0
│   │   │   │   │   │   ├── edgelet-http-workload 0.1.0
│   │   │   │   │   │   ├── edgelet-docker 0.1.0
│   │   │   │   │   │   └── aziot-edged 0.1.0
│   │   │   │   │   └── aziot-edged 0.1.0
│   │   │   │   ├── edgelet-docker 0.1.0
│   │   │   │   └── aziot-edged 0.1.0
│   │   │   ├── edgelet-http-workload 0.1.0
│   │   │   ├── edgelet-http-mgmt 0.1.0
│   │   │   ├── edgelet-http 0.1.0
│   │   │   ├── edgelet-docker 0.1.0
│   │   │   ├── cert-client 0.1.0
│   │   │   └── aziot-edged 0.1.0
│   │   └── aziot-edged 0.1.0
│   └── hmac 0.5.0
│       └── edgelet-core 0.1.0
└── crypto-mac 0.5.2
    └── hmac 0.5.0

Crate:         hyper
Version:       0.12.35
Title:         Multiple Transfer-Encoding headers misinterprets request payload
Date:          2021-02-05
ID:            RUSTSEC-2021-0020
URL:           https://rustsec.org/advisories/RUSTSEC-2021-0020
Solution:      Upgrade to >=0.14.3 OR >=0.13.10, <0.14.0 OR >=0.12.36, <0.13.0
Dependency tree:
hyper 0.12.35
├── workload 0.1.0
│   └── edgelet-http-workload 0.1.0
│       └── aziot-edged 0.1.0
├── management 0.1.0
│   ├── iotedge 0.1.0
│   └── edgelet-http-mgmt 0.1.0
│       ├── iotedge 0.1.0
│       ├── edgelet-http-workload 0.1.0
│       └── aziot-edged 0.1.0
├── iotedge 0.1.0
├── identity-client 0.1.0
│   ├── iotedge 0.1.0
│   ├── edgelet-http-workload 0.1.0
│   ├── edgelet-http-mgmt 0.1.0
│   └── aziot-edged 0.1.0
├── hyperlocal 0.6.0
│   ├── identity-client 0.1.0
│   ├── edgelet-test-utils 0.1.0
│   │   ├── support-bundle 0.1.0
│   │   │   ├── iotedge 0.1.0
│   │   │   └── edgelet-http-mgmt 0.1.0
│   │   ├── iotedge 0.1.0
│   │   ├── edgelet-http-workload 0.1.0
│   │   ├── edgelet-http-mgmt 0.1.0
│   │   ├── edgelet-http 0.1.0
│   │   │   ├── iotedge 0.1.0
│   │   │   ├── identity-client 0.1.0
│   │   │   ├── edgelet-http-workload 0.1.0
│   │   │   ├── edgelet-http-mgmt 0.1.0
│   │   │   ├── edgelet-docker 0.1.0
│   │   │   │   ├── iotedge 0.1.0
│   │   │   │   ├── edgelet-http-mgmt 0.1.0
│   │   │   │   └── aziot-edged 0.1.0
│   │   │   ├── cert-client 0.1.0
│   │   │   │   ├── edgelet-http-workload 0.1.0
│   │   │   │   ├── edgelet-docker 0.1.0
│   │   │   │   └── aziot-edged 0.1.0
│   │   │   └── aziot-edged 0.1.0
│   │   ├── edgelet-docker 0.1.0
│   │   └── aziot-edged 0.1.0
│   └── edgelet-http 0.1.0
├── hyper-tls 0.3.2
│   ├── iotedge 0.1.0
│   ├── hyper-proxy 0.5.1
│   │   ├── iotedge 0.1.0
│   │   └── edgelet-http 0.1.0
│   ├── edgelet-http 0.1.0
│   └── aziot-edged 0.1.0
├── hyper-proxy 0.5.1
├── edgelet-test-utils 0.1.0
├── edgelet-http-workload 0.1.0
├── edgelet-http-mgmt 0.1.0
├── edgelet-http 0.1.0
├── edgelet-docker 0.1.0
├── docker 0.1.0
│   ├── iotedge 0.1.0
│   ├── edgelet-docker 0.1.0
│   └── aziot-edged 0.1.0
├── cert-client 0.1.0
└── aziot-edged 0.1.0

Crate:         parse_duration
Version:       2.1.0
Title:         Denial of service through parsing payloads with too big exponent
Date:          2021-03-18
ID:            RUSTSEC-2021-0041
URL:           https://rustsec.org/advisories/RUSTSEC-2021-0041
Solution:      No safe upgrade is available!
Dependency tree:
parse_duration 2.1.0
└── edgelet-core 0.1.0
    ├── support-bundle 0.1.0
    │   ├── iotedge 0.1.0
    │   └── edgelet-http-mgmt 0.1.0
    │       ├── iotedge 0.1.0
    │       ├── edgelet-http-workload 0.1.0
    │       │   └── aziot-edged 0.1.0
    │       └── aziot-edged 0.1.0
    ├── iotedge 0.1.0
    ├── identity-client 0.1.0
    │   ├── iotedge 0.1.0
    │   ├── edgelet-http-workload 0.1.0
    │   ├── edgelet-http-mgmt 0.1.0
    │   └── aziot-edged 0.1.0
    ├── edgelet-test-utils 0.1.0
    │   ├── support-bundle 0.1.0
    │   ├── iotedge 0.1.0
    │   ├── edgelet-http-workload 0.1.0
    │   ├── edgelet-http-mgmt 0.1.0
    │   ├── edgelet-http 0.1.0
    │   │   ├── iotedge 0.1.0
    │   │   ├── identity-client 0.1.0
    │   │   ├── edgelet-http-workload 0.1.0
    │   │   ├── edgelet-http-mgmt 0.1.0
    │   │   ├── edgelet-docker 0.1.0
    │   │   │   ├── iotedge 0.1.0
    │   │   │   ├── edgelet-http-mgmt 0.1.0
    │   │   │   └── aziot-edged 0.1.0
    │   │   ├── cert-client 0.1.0
    │   │   │   ├── edgelet-http-workload 0.1.0
    │   │   │   ├── edgelet-docker 0.1.0
    │   │   │   └── aziot-edged 0.1.0
    │   │   └── aziot-edged 0.1.0
    │   ├── edgelet-docker 0.1.0
    │   └── aziot-edged 0.1.0
    ├── edgelet-http-workload 0.1.0
    ├── edgelet-http-mgmt 0.1.0
    ├── edgelet-http 0.1.0
    ├── edgelet-docker 0.1.0
    ├── cert-client 0.1.0
    └── aziot-edged 0.1.0

Crate:         smallvec
Version:       0.6.13
Title:         Buffer overflow in SmallVec::insert_many
Date:          2021-01-08
ID:            RUSTSEC-2021-0003
URL:           https://rustsec.org/advisories/RUSTSEC-2021-0003
Solution:      Upgrade to >=0.6.14, <1.0.0 OR >=1.6.1
Dependency tree:
smallvec 0.6.13
└── parking_lot_core 0.6.2
    └── parking_lot 0.9.0
        └── tokio-reactor 0.1.12
            ├── tokio-uds 0.2.7
            │   ├── tokio 0.1.22
            │   ├── hyperlocal 0.6.0
            │   │   ├── identity-client 0.1.0
            │   │   │   ├── iotedge 0.1.0
            │   │   │   ├── edgelet-http-workload 0.1.0
            │   │   │   │   └── aziot-edged 0.1.0
            │   │   │   ├── edgelet-http-mgmt 0.1.0
            │   │   │   │   ├── iotedge 0.1.0
            │   │   │   │   ├── edgelet-http-workload 0.1.0
            │   │   │   │   └── aziot-edged 0.1.0
            │   │   │   └── aziot-edged 0.1.0
            │   │   ├── edgelet-test-utils 0.1.0
            │   │   │   ├── support-bundle 0.1.0
            │   │   │   │   ├── iotedge 0.1.0
            │   │   │   │   └── edgelet-http-mgmt 0.1.0
            │   │   │   ├── iotedge 0.1.0
            │   │   │   ├── edgelet-http-workload 0.1.0
            │   │   │   ├── edgelet-http-mgmt 0.1.0
            │   │   │   ├── edgelet-http 0.1.0
            │   │   │   │   ├── iotedge 0.1.0
            │   │   │   │   ├── identity-client 0.1.0
            │   │   │   │   ├── edgelet-http-workload 0.1.0
            │   │   │   │   ├── edgelet-http-mgmt 0.1.0
            │   │   │   │   ├── edgelet-docker 0.1.0
            │   │   │   │   │   ├── iotedge 0.1.0
            │   │   │   │   │   ├── edgelet-http-mgmt 0.1.0
            │   │   │   │   │   └── aziot-edged 0.1.0
            │   │   │   │   ├── cert-client 0.1.0
            │   │   │   │   │   ├── edgelet-http-workload 0.1.0
            │   │   │   │   │   ├── edgelet-docker 0.1.0
            │   │   │   │   │   └── aziot-edged 0.1.0
            │   │   │   │   └── aziot-edged 0.1.0
            │   │   │   ├── edgelet-docker 0.1.0
            │   │   │   └── aziot-edged 0.1.0
            │   │   └── edgelet-http 0.1.0
            │   └── edgelet-http 0.1.0
            ├── tokio-udp 0.1.6
            │   └── tokio 0.1.22
            ├── tokio-tcp 0.1.4
            │   ├── tokio 0.1.22
            │   └── hyper 0.12.35
            │       ├── workload 0.1.0
            │       │   └── edgelet-http-workload 0.1.0
            │       ├── management 0.1.0
            │       │   ├── iotedge 0.1.0
            │       │   └── edgelet-http-mgmt 0.1.0
            │       ├── iotedge 0.1.0
            │       ├── identity-client 0.1.0
            │       ├── hyperlocal 0.6.0
            │       ├── hyper-tls 0.3.2
            │       │   ├── iotedge 0.1.0
            │       │   ├── hyper-proxy 0.5.1
            │       │   │   ├── iotedge 0.1.0
            │       │   │   └── edgelet-http 0.1.0
            │       │   ├── edgelet-http 0.1.0
            │       │   └── aziot-edged 0.1.0
            │       ├── hyper-proxy 0.5.1
            │       ├── edgelet-test-utils 0.1.0
            │       ├── edgelet-http-workload 0.1.0
            │       ├── edgelet-http-mgmt 0.1.0
            │       ├── edgelet-http 0.1.0
            │       ├── edgelet-docker 0.1.0
            │       ├── docker 0.1.0
            │       │   ├── iotedge 0.1.0
            │       │   ├── edgelet-docker 0.1.0
            │       │   └── aziot-edged 0.1.0
            │       ├── cert-client 0.1.0
            │       └── aziot-edged 0.1.0
            ├── tokio-signal 0.2.9
            │   ├── tokio-process 0.2.5
            │   │   └── edgelet-docker 0.1.0
            │   └── aziot-edged 0.1.0
            ├── tokio-process 0.2.5
            ├── tokio 0.1.22
            └── hyper 0.12.35

Crate:         failure
Version:       0.1.8
Warning:       unmaintained
Title:         failure is officially deprecated/unmaintained
Date:          2020-05-02
ID:            RUSTSEC-2020-0036
URL:           https://rustsec.org/advisories/RUSTSEC-2020-0036
Dependency tree:
failure 0.1.8
├── workload 0.1.0
│   └── edgelet-http-workload 0.1.0
│       └── aziot-edged 0.1.0
├── systemd 0.1.0
│   └── edgelet-http 0.1.0
│       ├── iotedge 0.1.0
│       ├── identity-client 0.1.0
│       │   ├── iotedge 0.1.0
│       │   ├── edgelet-http-workload 0.1.0
│       │   ├── edgelet-http-mgmt 0.1.0
│       │   │   ├── iotedge 0.1.0
│       │   │   ├── edgelet-http-workload 0.1.0
│       │   │   └── aziot-edged 0.1.0
│       │   └── aziot-edged 0.1.0
│       ├── edgelet-http-workload 0.1.0
│       ├── edgelet-http-mgmt 0.1.0
│       ├── edgelet-docker 0.1.0
│       │   ├── iotedge 0.1.0
│       │   ├── edgelet-http-mgmt 0.1.0
│       │   └── aziot-edged 0.1.0
│       ├── cert-client 0.1.0
│       │   ├── edgelet-http-workload 0.1.0
│       │   ├── edgelet-docker 0.1.0
│       │   └── aziot-edged 0.1.0
│       └── aziot-edged 0.1.0
├── support-bundle 0.1.0
│   ├── iotedge 0.1.0
│   └── edgelet-http-mgmt 0.1.0
├── management 0.1.0
│   ├── iotedge 0.1.0
│   └── edgelet-http-mgmt 0.1.0
├── iotedge 0.1.0
├── identity-client 0.1.0
├── edgelet-utils 0.1.0
│   ├── iotedge 0.1.0
│   ├── edgelet-http-workload 0.1.0
│   ├── edgelet-http 0.1.0
│   ├── edgelet-docker 0.1.0
│   ├── edgelet-core 0.1.0
│   │   ├── support-bundle 0.1.0
│   │   ├── iotedge 0.1.0
│   │   ├── identity-client 0.1.0
│   │   ├── edgelet-test-utils 0.1.0
│   │   │   ├── support-bundle 0.1.0
│   │   │   ├── iotedge 0.1.0
│   │   │   ├── edgelet-http-workload 0.1.0
│   │   │   ├── edgelet-http-mgmt 0.1.0
│   │   │   ├── edgelet-http 0.1.0
│   │   │   ├── edgelet-docker 0.1.0
│   │   │   └── aziot-edged 0.1.0
│   │   ├── edgelet-http-workload 0.1.0
│   │   ├── edgelet-http-mgmt 0.1.0
│   │   ├── edgelet-http 0.1.0
│   │   ├── edgelet-docker 0.1.0
│   │   ├── cert-client 0.1.0
│   │   └── aziot-edged 0.1.0
│   └── aziot-edged 0.1.0
├── edgelet-test-utils 0.1.0
├── edgelet-http-workload 0.1.0
├── edgelet-http-mgmt 0.1.0
├── edgelet-http 0.1.0
├── edgelet-docker 0.1.0
├── edgelet-core 0.1.0
├── docker 0.1.0
│   ├── iotedge 0.1.0
│   ├── edgelet-docker 0.1.0
│   └── aziot-edged 0.1.0
├── cert-client 0.1.0
└── aziot-edged 0.1.0

Crate:         net2
Version:       0.2.35
Warning:       unmaintained
Title:         `net2` crate has been deprecated; use `socket2` instead
Date:          2020-05-01
ID:            RUSTSEC-2020-0016
URL:           https://rustsec.org/advisories/RUSTSEC-2020-0016
Dependency tree:
net2 0.2.35
├── miow 0.2.1
├── mio 0.6.22
│   ├── tokio-uds 0.2.7
│   │   ├── tokio 0.1.22
│   │   ├── hyperlocal 0.6.0
│   │   │   ├── identity-client 0.1.0
│   │   │   │   ├── iotedge 0.1.0
│   │   │   │   ├── edgelet-http-workload 0.1.0
│   │   │   │   │   └── aziot-edged 0.1.0
│   │   │   │   ├── edgelet-http-mgmt 0.1.0
│   │   │   │   │   ├── iotedge 0.1.0
│   │   │   │   │   ├── edgelet-http-workload 0.1.0
│   │   │   │   │   └── aziot-edged 0.1.0
│   │   │   │   └── aziot-edged 0.1.0
│   │   │   ├── edgelet-test-utils 0.1.0
│   │   │   │   ├── support-bundle 0.1.0
│   │   │   │   │   ├── iotedge 0.1.0
│   │   │   │   │   └── edgelet-http-mgmt 0.1.0
│   │   │   │   ├── iotedge 0.1.0
│   │   │   │   ├── edgelet-http-workload 0.1.0
│   │   │   │   ├── edgelet-http-mgmt 0.1.0
│   │   │   │   ├── edgelet-http 0.1.0
│   │   │   │   │   ├── iotedge 0.1.0
│   │   │   │   │   ├── identity-client 0.1.0
│   │   │   │   │   ├── edgelet-http-workload 0.1.0
│   │   │   │   │   ├── edgelet-http-mgmt 0.1.0
│   │   │   │   │   ├── edgelet-docker 0.1.0
│   │   │   │   │   │   ├── iotedge 0.1.0
│   │   │   │   │   │   ├── edgelet-http-mgmt 0.1.0
│   │   │   │   │   │   └── aziot-edged 0.1.0
│   │   │   │   │   ├── cert-client 0.1.0
│   │   │   │   │   │   ├── edgelet-http-workload 0.1.0
│   │   │   │   │   │   ├── edgelet-docker 0.1.0
│   │   │   │   │   │   └── aziot-edged 0.1.0
│   │   │   │   │   └── aziot-edged 0.1.0
│   │   │   │   ├── edgelet-docker 0.1.0
│   │   │   │   └── aziot-edged 0.1.0
│   │   │   └── edgelet-http 0.1.0
│   │   └── edgelet-http 0.1.0
│   ├── tokio-udp 0.1.6
│   │   └── tokio 0.1.22
│   ├── tokio-tcp 0.1.4
│   │   ├── tokio 0.1.22
│   │   └── hyper 0.12.35
│   │       ├── workload 0.1.0
│   │       │   └── edgelet-http-workload 0.1.0
│   │       ├── management 0.1.0
│   │       │   ├── iotedge 0.1.0
│   │       │   └── edgelet-http-mgmt 0.1.0
│   │       ├── iotedge 0.1.0
│   │       ├── identity-client 0.1.0
│   │       ├── hyperlocal 0.6.0
│   │       ├── hyper-tls 0.3.2
│   │       │   ├── iotedge 0.1.0
│   │       │   ├── hyper-proxy 0.5.1
│   │       │   │   ├── iotedge 0.1.0
│   │       │   │   └── edgelet-http 0.1.0
│   │       │   ├── edgelet-http 0.1.0
│   │       │   └── aziot-edged 0.1.0
│   │       ├── hyper-proxy 0.5.1
│   │       ├── edgelet-test-utils 0.1.0
│   │       ├── edgelet-http-workload 0.1.0
│   │       ├── edgelet-http-mgmt 0.1.0
│   │       ├── edgelet-http 0.1.0
│   │       ├── edgelet-docker 0.1.0
│   │       ├── docker 0.1.0
│   │       │   ├── iotedge 0.1.0
│   │       │   ├── edgelet-docker 0.1.0
│   │       │   └── aziot-edged 0.1.0
│   │       ├── cert-client 0.1.0
│   │       └── aziot-edged 0.1.0
│   ├── tokio-signal 0.2.9
│   │   ├── tokio-process 0.2.5
│   │   │   └── edgelet-docker 0.1.0
│   │   └── aziot-edged 0.1.0
│   ├── tokio-reactor 0.1.12
│   │   ├── tokio-uds 0.2.7
│   │   ├── tokio-udp 0.1.6
│   │   ├── tokio-tcp 0.1.4
│   │   ├── tokio-signal 0.2.9
│   │   ├── tokio-process 0.2.5
│   │   ├── tokio 0.1.22
│   │   └── hyper 0.12.35
│   ├── tokio-process 0.2.5
│   ├── tokio 0.1.22
│   ├── mio-uds 0.6.8
│   │   ├── tokio-uds 0.2.7
│   │   └── tokio-signal 0.2.9
│   └── mio-named-pipes 0.1.7
│       └── tokio-process 0.2.5
└── hyper 0.12.35

Crate:         tempdir
Version:       0.3.7
Warning:       unmaintained
Title:         `tempdir` crate has been deprecated; use `tempfile` instead
Date:          2018-02-13
ID:            RUSTSEC-2018-0017
URL:           https://rustsec.org/advisories/RUSTSEC-2018-0017
Dependency tree:
tempdir 0.3.7
├── edgelet-http 0.1.0
│   ├── iotedge 0.1.0
│   ├── identity-client 0.1.0
│   │   ├── iotedge 0.1.0
│   │   ├── edgelet-http-workload 0.1.0
│   │   │   └── aziot-edged 0.1.0
│   │   ├── edgelet-http-mgmt 0.1.0
│   │   │   ├── iotedge 0.1.0
│   │   │   ├── edgelet-http-workload 0.1.0
│   │   │   └── aziot-edged 0.1.0
│   │   └── aziot-edged 0.1.0
│   ├── edgelet-http-workload 0.1.0
│   ├── edgelet-http-mgmt 0.1.0
│   ├── edgelet-docker 0.1.0
│   │   ├── iotedge 0.1.0
│   │   ├── edgelet-http-mgmt 0.1.0
│   │   └── aziot-edged 0.1.0
│   ├── cert-client 0.1.0
│   │   ├── edgelet-http-workload 0.1.0
│   │   ├── edgelet-docker 0.1.0
│   │   └── aziot-edged 0.1.0
│   └── aziot-edged 0.1.0
├── edgelet-docker 0.1.0
└── aziot-edged 0.1.0

Crate:         arc-swap
Version:       0.4.7
Warning:       yanked

Crate:         miow
Version:       0.2.1
Warning:       yanked
Dependency tree:
miow 0.2.1

Crate:         miow
Version:       0.3.5
Warning:       yanked
Dependency tree:
miow 0.3.5

Crate:         net2
Version:       0.2.35
Warning:       yanked

Crate:         pin-project-lite
Version:       0.2.4
Warning:       yanked
Dependency tree:
pin-project-lite 0.2.4
├── tracing 0.1.25
│   └── http-common 0.1.0
│       ├── edgelet-http-workload 0.1.0
│       │   └── aziot-edged 0.1.0
│       ├── aziotctl-common 0.1.0
│       │   └── iotedge 0.1.0
│       ├── aziot-tpmd-config 0.1.0
│       │   ├── iotedge 0.1.0
│       │   └── aziotctl-common 0.1.0
│       ├── aziot-keyd-config 0.1.0
│       │   ├── iotedge 0.1.0
│       │   └── aziotctl-common 0.1.0
│       ├── aziot-key-common-http 0.1.0
│       │   ├── edgelet-http-workload 0.1.0
│       │   ├── aziot-key-client 0.1.0
│       │   │   ├── edgelet-http-workload 0.1.0
│       │   │   ├── aziot-key-openssl-engine 0.1.0
│       │   │   │   ├── edgelet-http-workload 0.1.0
│       │   │   │   └── aziot-edged 0.1.0
│       │   │   └── aziot-edged 0.1.0
│       │   └── aziot-edged 0.1.0
│       ├── aziot-key-client 0.1.0
│       ├── aziot-identityd-config 0.1.0
│       │   ├── iotedge 0.1.0
│       │   └── aziotctl-common 0.1.0
│       ├── aziot-identity-common-http 0.1.0
│       │   ├── iotedge 0.1.0
│       │   ├── identity-client 0.1.0
│       │   │   ├── iotedge 0.1.0
│       │   │   ├── edgelet-http-workload 0.1.0
│       │   │   ├── edgelet-http-mgmt 0.1.0
│       │   │   │   ├── iotedge 0.1.0
│       │   │   │   ├── edgelet-http-workload 0.1.0
│       │   │   │   └── aziot-edged 0.1.0
│       │   │   └── aziot-edged 0.1.0
│       │   └── aziot-edged 0.1.0
│       ├── aziot-identity-common 0.1.0
│       │   ├── iotedge 0.1.0
│       │   ├── identity-client 0.1.0
│       │   ├── edgelet-http-workload 0.1.0
│       │   ├── edgelet-http-mgmt 0.1.0
│       │   ├── aziot-identityd-config 0.1.0
│       │   ├── aziot-identity-common-http 0.1.0
│       │   └── aziot-edged 0.1.0
│       ├── aziot-edged 0.1.0
│       └── aziot-certd-config 0.1.0
│           ├── iotedge 0.1.0
│           └── aziotctl-common 0.1.0
├── tokio-io-timeout 1.1.1
│   └── http-common 0.1.0
└── tokio 1.5.0

Crate:         socket2
Version:       0.3.12
Warning:       yanked
Dependency tree:
socket2 0.3.12
└── miow 0.3.5

error: 7 vulnerabilities found!
warning: 9 allowed warnings found
```

**Comment by nyanzebra** [comment]
I think it might also be worthwhile to add cargo audit to build pipeline. Maybe in a separate pr... would there be any concerns with doing this?

**Comment by arsing** [comment]
(You can put walls of text in a `details` element. See what I edited your post to.)

**Review by arsing — APPROVED** [review]

**Pull request merged**